api/ocm/compdesc/meta/v1/identity.go

```diff
@@ -103,6 +103,22 @@ func (i Identity) Remove(name string) bool {
 	return false
 }
 
+// ExtraIdentity extracts the extra identity part of an identity.
+func (i Identity) ExtraIdentity() Identity {
+	if i == nil {
+		return nil
+	}
+	if _, ok := i[SystemIdentityName]; !ok {
+		return i
+	}
+	if len(i) == 1 {
+		return nil
+	}
+	r := i.Copy()
+	delete(r, SystemIdentityName)
```

Comment:
These two operations together cost more than just simply looping through the map and creating a new one but skipping the system identity key.

Reply:
yes, may be, but you have to implement it completely on your side, by copying the copy code, to be performant.